mtp_api/apps/security/tests/test_views/test_check_views.py

```diff
@@ -0,0 +1,1231 @@
+        response = self.client.patch(
+            reverse(
+                'security-check-detail',
+                kwargs={'pk': check.pk},
+            ),
+            data={
+                'assigned_to': assigned_to_user.id
+            },
+            format='json',
+            HTTP_AUTHORIZATION=auth,
+        )
+        self.assertEqual(response.status_code, http_status.HTTP_200_OK)
+        actual_check_data = response.json()
+
+        assert check.status != CHECK_STATUS.ACCEPTED
+        response = self.client.patch(
+            reverse(
+                'security-check-detail',
+                kwargs={'pk': check.pk},
+            ),
+            data={
+                'status': CHECK_STATUS.ACCEPTED
+            },
+            format='json',
+            HTTP_AUTHORIZATION=auth,
+        )
+
+        # Assert check unchanged
+        self.assertEqual(actual_check_data['assigned_to'], assigned_to_user.id)
+        self.assertEqual(actual_check_data['assigned_to_name'], assigned_to_user.get_full_name())
+
+        check = Check.objects.get(pk=actual_check_data['id'])
+        self.assertCheckEqual(check, actual_check_data)
+
+
+class GetCheckTestCase(BaseCheckTestCase):
+    """
+    Tests related to getting a single security check.
+    """
+
+    def test_get(self):
+        """
+        Test that the get object endpoint returns check details.
+        """
+        check = Check.objects.first()
+
+        auth = self.get_http_authorization_for_user(self._get_authorised_user())
+        response = self.client.get(
+            reverse(
+                'security-check-detail',
+                kwargs={'pk': check.pk},
+            ),
+            format='json',
+            HTTP_AUTHORIZATION=auth,
+        )
+        self.assertEqual(response.status_code, http_status.HTTP_200_OK)
+
+        actual_check_data = response.json()
+        self.assertCheckEqual(check, actual_check_data)
+
+    def test_get_assigned_check(self):
+        """
+        Test that the get object endpoint returns check details including assignment
+        """
+        check = Check.objects.first()
+        check.assigned_to = self.security_fiu_users[0]
+        check.save()
+
+        auth = self.get_http_authorization_for_user(self._get_authorised_user())
+        response = self.client.get(
+            reverse(
+                'security-check-detail',
+                kwargs={'pk': check.pk},
+            ),
+            format='json',
+            HTTP_AUTHORIZATION=auth,
+        )
+        self.assertEqual(response.status_code, http_status.HTTP_200_OK)
+
+        actual_check_data = response.json()
+        self.assertCheckEqual(check, actual_check_data)
+
+    def test_get_check_auto_accept_rule_state_attached(self):
+        """
+        Test that the get object endpoint returns CheckAutoAcceptRuleState when appropriate
+        """
+        # Setup
+        response = self.client.post(
```

Comment:
to me, it's a little self-referential to use a view to create test data in order to test other views. in other places we either use the model or mommy creation methods directly. but really i don't mind since this data-creating view itself is tested elsewhere in isolation from other views.

Reply:
I see what you're saying, my thinking is that it's not worth adding other test setup factory that we're only going to use in this class, and if we're going to set up the data somehow, then it's it's nice and explicit to use the actual API